_test/test_sqw/test_PixelData.m

```diff
@@ -1113,6 +1114,208 @@ function test_PIXELDATA_raised_if_mem_alloc_argument_is_not_scalar(~)
+        num_pix = 30;
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix);
+        npix_in_page = 11;
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        rand_order = randperm(num_pix);
+        pix_out = pix.get_pixels(rand_order);
+
+        assertEqual(pix_out.data, data(:, rand_order));
+    end
+
+    function test_get_pixels_throws_invalid_arg_if_indices_not_vector(~)
+        pix = PixelData();
+        f = @() pix.get_pixels(ones(2, 2));
+        assertExceptionThrown(f, 'MATLAB:InputParser:ArgumentFailedValidation');
+    end
+
+    function test_get_pixels_throws_if_range_out_of_bounds(obj)
+        num_pix = 30;
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix);
+        npix_in_page = 11;
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        idx_array = 25:35;
+        f = @() pix.get_pixels(idx_array);
+        assertExceptionThrown(f, 'PIXELDATA:get_pixels');
+    end
+
+    function test_get_pixels_throws_if_an_idx_lt_1_with_paged_pix(obj)
+        num_pix = 30;
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix);
+        npix_in_page = 11;
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        idx_array = -1:20;
+        f = @() pix.get_pixels(idx_array);
+        assertExceptionThrown(f, 'MATLAB:InputParser:ArgumentFailedValidation');
+    end
+
+    function test_get_pixels_throws_if_indices_not_positive_int(~)
+        pix = PixelData();
+        idx_array = 1:0.1:5;
+        f = @() pix.get_pixels(idx_array);
+        assertExceptionThrown(f, 'MATLAB:InputParser:ArgumentFailedValidation');
+    end
+
+    function test_paged_pix_get_pixels_can_be_called_with_a_logical(obj)
+        num_pix = 30;
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix);
+        npix_in_page = 11;
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        logical_array = logical(randi([0, 1], [1, 10]));
+        pix_out = pix.get_pixels(logical_array);
+
+        assertEqual(pix_out.data, data(:, logical_array));
+    end
+
+    function test_in_mem_pix_get_pixels_can_be_called_with_a_logical(~)
+        num_pix = 30;
+        pix = PixelData(rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix));
+
+        logical_array = logical(randi([0, 1], [1, 10]));
+        pix_out = pix.get_pixels(logical_array);
+
+        assertEqual(pix_out.data, pix.data(:, logical_array));
+    end
+
+    function test_get_pixels_can_handle_repeated_indices(obj)
+        num_pix = 30;
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix);
+        npix_in_page = 11;
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        idx_array = cat(2, randperm(num_pix), randperm(num_pix));
+
+        pix_chunk = pix.get_pixels(idx_array);
+        assertEqual(pix_chunk.data, data(:, idx_array));
+    end
+
+    function test_pg_size_is_correct_after_advance(obj)
```

Comment:
is the test here that "page size reports size of partially full final page" rather than "correct after advance"